drivers/display/display_dummy.c

```diff
@@ -29,6 +29,15 @@ static int dummy_display_write(const struct device *dev, const uint16_t x,
 			       const struct display_buffer_descriptor *desc,
 			       const void *buf)
 {
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT(desc->pitch <= CONFIG_DUMMY_DISPLAY_X_RES,
+		"Pitch in descriptor is larger than screen size");
+	__ASSERT(desc->height <= CONFIG_DUMMY_DISPLAY_Y_RES,
+		"Height in descriptor is larger than screen size");
+	__ASSERT(x + desc->pitch <= CONFIG_DUMMY_DISPLAY_X_RES,
+		 "Writing outside screen boundaries in horizontal direction");
+	__ASSERT(y + desc->height <= CONFIG_DUMMY_DISPLAY_Y_RES,
+		 "Writing outside screen boundaries in vertical direction");
```

Comment:
I think that display API should clarify what runtime conditions are allowed/forbidden. E.g. when writing outside boundaries, should it be an assertion or an error return (`EINVAL, ERANGE`)? All drivers should exhibit the same behavior.

Reply:
Not sure if it needs to be enforced by the API, I think it is up to each driver to handle this case. Could well be that display controller has no issue with out of bound writes.

On the other hand as this is a test driver I will return `ERANGE` next to the assertion.